planning/behavior_path_planner/src/scene_module/lane_change/avoidance_by_lane_change.cpp

```diff
@@ -34,13 +34,26 @@ AvoidanceByLaneChange::AvoidanceByLaneChange(
 bool AvoidanceByLaneChange::specialRequiredCheck() const
 {
   const auto & data = avoidance_data_;
-  const auto & execute_object_num = avoidance_parameters_->execute_object_num;
 
   if (!status_.is_safe) {
     return false;
   }
 
-  if (data.target_objects.size() < execute_object_num) {
+  const auto & objects = avoidance_data_.target_objects;
+
+  bool is_more_than_threshold = false;
+  for (const auto & p : avoidance_parameters_->object_parameters) {
+    const size_t num = std::count_if(objects.begin(), objects.end(), [&p](const auto & object) {
+      const auto target_class = utils::getHighestProbLabel(object.object.classification) == p.first;
+      return target_class && object.avoid_required;
+    });
+
+    if (num >= p.second.execute_num) {
+      is_more_than_threshold = true;
+    }
+  }
+
+  if (!is_more_than_threshold) {
```

Comment:
```suggestion
  const auto is_less_than_threshold =
    std::any_of(object_parameters.begin(), object_parameters.end(), [&](const auto & p) {
      const auto & objects = avoidance_data_.target_objects;
      const size_t num = std::count_if(objects.begin(), objects.end(), [&p](const auto & object) {
        const auto target_class =
          utils::getHighestProbLabel(object.object.classification) == p.first;
        return target_class && object.avoid_required;
      });
      return num < p.second.execute;
    });

  if (!is_less_than_threshold) {
```

@satoshi-ota 
This part over here seems suitable to use `std::any_of`, so this is my suggestion (I am sorry for changing some part). 
Looking forward to hear you thought in this.

Reply:
Fixed in https://github.com/autowarefoundation/autoware.universe/pull/4258/commits/2e2c965a02c4a61cd80818e1565be6f0a707c88c.